zhaquirks/tuya/ts0601_trv_sas.py

```diff
@@ -81,6 +85,25 @@ def _update_attribute(self, attrid, value):
             self.endpoint.device.thermostat_bus.listener_event(
                 LOCAL_TEMP_REPORTED, value
             )
+        elif attrid == SCHEDULE_ENABLED_COMMAND_ID:
+            self.endpoint.device.thermostat_bus.listener_event(
+                SCHEDULE_STATE_REPORTED, value
+            )
+
+    async def write_attributes(self, attributes, manufacturer=None):
+        """Implement writeable attributes."""
+
+        if "system_mode" in attributes:
+
+            async def _disable_schedule():
+                await asyncio.sleep(3)
+                await self.write_attributes(
+                    {"schedule_enabled": 0}, manufacturer=manufacturer
+                )
+
+            asyncio.get_running_loop().create_task(_disable_schedule())
```

Comment:
Maybe you can do the same with:
```python
        if "system_mode" in attributes:
            await self.write_attributes({"schedule_enabled": 0}, manufacturer=manufacturer)

        return await super().write_attributes(attributes, manufacturer)
```

(Despite I don't know the relationship between the `system_mode` and `schedule_enabled` attributes) 🤷🏻‍♂️

Reply:
Changed to send state and schedule mode simultaneously.